cpp/src/gandiva/engine.cc

```diff
@@ -454,10 +555,9 @@ arrow::Status Engine::AddGlobalMappings() {
 }
 
 std::string Engine::DumpIR() {
```

Comment:
```suggestion
const std::string& Engine::ir() {
```

Reply:
Updated. But I still keep the API in `Filter` and `Projector` as `DumpIR` since they may be used by apps already (but change them to return `const string&`

Reply:
OK. How about adding new `const std::string& Engine::ir()` and keeping `ARROW_DEPRECATED("Deprecated in 15.0.0. Use ir() instead.") std::string Engine:::DumpIR() {return ir();}` for backward compatibility?